ccm-connector/src/main/java/com/sequenceiq/cloudbreak/ccmimpl/altus/MinaSshdManagementClient.java

```diff
@@ -0,0 +1,214 @@
+import com.cloudera.thunderhead.service.minasshdmanagement.MinaSshdManagementProto.AcquireMinaSshdServiceRequest;
+import com.cloudera.thunderhead.service.minasshdmanagement.MinaSshdManagementProto.AcquireMinaSshdServiceResponse;
+import com.cloudera.thunderhead.service.minasshdmanagement.MinaSshdManagementProto.GenerateAndRegisterSshTunnelingKeyPairRequest;
+import com.cloudera.thunderhead.service.minasshdmanagement.MinaSshdManagementProto.GenerateAndRegisterSshTunnelingKeyPairResponse;
+import com.cloudera.thunderhead.service.minasshdmanagement.MinaSshdManagementProto.ListMinaSshdServicesRequest;
+import com.cloudera.thunderhead.service.minasshdmanagement.MinaSshdManagementProto.ListMinaSshdServicesResponse;
+import com.cloudera.thunderhead.service.minasshdmanagement.MinaSshdManagementProto.MinaSshdService;
+import com.sequenceiq.cloudbreak.ccm.exception.CcmException;
+import com.sequenceiq.cloudbreak.ccmimpl.altus.config.MinaSshdManagementClientConfig;
+import com.sequenceiq.cloudbreak.grpc.altus.AltusMetadataInterceptor;
+import com.sequenceiq.cloudbreak.grpc.util.GrpcUtil;
+
+import io.grpc.ManagedChannel;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+
+/**
+ * A wrapper to the GRPC minasshd management service, that handles setting up the appropriate context-propagating interceptors,
+ * hides some boilerplate, and takes care of retries.
+ */
+public class MinaSshdManagementClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MinaSshdManagementClient.class);
+
+    private final ManagedChannel channel;
+
+    private final String actorCrn;
+
+    private final MinaSshdManagementClientConfig minaSshdManagementClientConfig;
+
+    /**
+     * Constructor.
+     *
+     * @param channel the managed channel.
+     */
+    MinaSshdManagementClient(ManagedChannel channel, String actorCrn,
+            MinaSshdManagementClientConfig minaSshdManagementClientConfig) {
+        this.channel = checkNotNull(channel, "channel is null");
+        this.actorCrn = checkNotNull(actorCrn, "actorCrn is null");
+        this.minaSshdManagementClientConfig = checkNotNull(minaSshdManagementClientConfig);
+    }
+
+    /**
+     * Wraps call to acquireMinaSshdService.
+     *
+     * @param requestId the request ID for the request
+     * @param accountId the account ID
+     * @return the minasshd service
+     * @throws CcmException if an exception occurs
+     */
+    public MinaSshdService acquireMinaSshdService(String requestId, String accountId) throws CcmException {
+        checkNotNull(requestId);
+        checkNotNull(accountId);
+
+        MinaSshdManagementBlockingStub blockingStub = newStub(requestId);
+        AcquireMinaSshdServiceRequest.Builder requestBuilder = AcquireMinaSshdServiceRequest.newBuilder()
+                .setAccountId(accountId);
+
+        try {
+            if (LOGGER.isDebugEnabled()) {
```

Comment:
the LOGGER.isDebugEnabled checks seem redundant since you aren't doing any work in your LOGGER.debug calls.

Reply:
k will change